src/main/java/eu/greev/dcbot/ticketsystem/interactions/commands/ThreadAdd.java

```diff
@@ -0,0 +1,79 @@
+package eu.greev.dcbot.ticketsystem.interactions.commands;
+
+import eu.greev.dcbot.ticketsystem.service.TicketService;
+import eu.greev.dcbot.utils.Config;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.channel.ChannelType;
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.events.Event;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+
+import java.awt.*;
+
+public class ThreadAdd extends AbstractCommand {
+    private final EmbedBuilder wrongChannel;
+
+    public ThreadAdd(Config config, TicketService ticketService, EmbedBuilder wrongChannel, EmbedBuilder missingPerm, JDA jda) {
+        super(config, ticketService, missingPerm, jda);
+        this.wrongChannel = wrongChannel;
+    }
+
+    @Override
+    public void execute(Event evt) {
+        SlashCommandInteractionEvent event = (SlashCommandInteractionEvent) evt;
+        if (!fromGuild(event)) return;
+
+        if (config.getServerName() == null) {
```

Comment:
I noticed this check appears multiple times, maybe you can make it into its own service or something that gets checked before the event gets sent to the commands in the first place. Same for the fromGuild check

Reply:
Yeah, but ig the Setup.java class is a Command where it doesnt matter if the system isnt setup, so i cant just check in the Listener before every execute if the Ticket system isnt setup